.vib/clickhouse/goss/goss.yaml

```
@@ -0,0 +1,42 @@
+file:
+  /opt/bitnami/clickhouse/etc/config.xml:
+    mode: "0664"
+    filetype: file
+    exists: true
+  /opt/bitnami/clickhouse/etc/users.xml:
+    mode: "0664"
+    filetype: file
+    exists: true
+  /bitnami/clickhouse/data:
+    mode: "2750"
+    filetype: directory
+    exists: true
+http:
+  http://clickhouse:{{ .Vars.service.ports.http }}:
+    status: 200
+    allow-insecure: true
+    body:
+      - Ok
+  http://clickhouse:{{ .Vars.service.ports.http }}/replicas_status:
+    status: 200
+    allow-insecure: true
+    body:
+      - Ok
+command:
+  user-id-test:
+    exec: if [ "$(id -u)" -eq 0 ]; then exit 1; fi
+    exit-status: 0
+    stdout: []
+    stderr: []
+  check-cluster-status:
+    exec: clickhouse-client --password '{{ .Vars.auth.password }}' --query='SELECT COUNT(*) FROM system.clusters;'
+    exit-status: 0
+    stdout:
+      - 4
+  create-database-in-cluster:
+    {{- $table := printf "%s.%s" .Vars.database .Vars.table }}
+    exec: clickhouse-client --multiquery --password '{{ .Vars.auth.password }}' --query='CREATE DATABASE {{ .Vars.database }} ON CLUSTER default ENGINE = Memory; CREATE TABLE {{ $table }} (a UInt8) ENGINE = Memory;  INSERT INTO {{ $table }} VALUES(23); SELECT * FROM {{ $table }};'
```

As a good practice, we are defining both Cypress and Goss tests for them to be auto-conclusive and retryable. In this case, additional executions would fail with the following error:

```
Code: 82. DB::Exception: Database test_db_goss already exists.
```

Usually, we use one of the following solutions:
- Start the test by removing the database if it exists
- Randomize some of the input data (preferred one).

In this case we could achieve the data randomization by doing:

```suggestion
  create-database-in-cluster:
    {{- $database := printf "test_db_%s" (randAlpha 5) }}
    {{- $table := printf "%s.test_table_%s" $database (randAlpha 5) }}
    exec: clickhouse-client --multiquery --password '{{ .Vars.auth.password }}' --query='CREATE DATABASE {{ $database }} ON CLUSTER default ENGINE = Memory; CREATE TABLE {{ $table }} (a UInt8) ENGINE = Memory;  INSERT INTO {{ $table }} VALUES(23); SELECT * FROM {{ $table }};'
```